src/SFA.DAS.EAS.Portal.Worker/EventHandlers/Commitments/CohortApprovalRequestedByProviderEventHandler.cs

```diff
@@ -0,0 +1,34 @@
+﻿using NServiceBus;
+using SFA.DAS.CommitmentsV2.Messages.Events;
+using SFA.DAS.EAS.Portal.Application.Adapters;
+using SFA.DAS.EAS.Portal.Application.Commands;
+using SFA.DAS.EAS.Portal.Application.Commands.Cohort;
+using SFA.DAS.EAS.Portal.Application.Services;
+using SFA.DAS.EAS.Portal.Worker.Extensions;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EAS.Portal.Worker.EventHandlers.Commitments
+{
+    public class CohortApprovalRequestedByProviderEventHandler : IHandleMessages<CohortApprovalRequestedByProvider>
+    {
+        private readonly IMessageContext _messageContext;
+        private readonly ICommandHandler<CohortApprovalRequestedCommand> _handler;
+        private readonly IAdapter<CohortApprovalRequestedByProvider, CohortApprovalRequestedCommand> _adapter;
+
+        public CohortApprovalRequestedByProviderEventHandler(
```

Comment:
Comparing the pattern of this event handler and command combo, to the pattern employed by `ReservationCreatedEventHandler` and its related command, I personally think the pattern used by the latter is simpler and easier to follow.

They both solve the same problem, but the pattern used by the cohort handler requires these extra components:
* a command, command handler and command handler interface, rather than just a command
* an adaptor and interface
* adding the above to the service collection, including scanning for command handlers

The pattern used by the cohort handler does have the advantage of decoupling the event handler and command handler, so really I think it comes down to the question of whether that is worth the extra complexity. I do think though, that there is value in developers being able to understand the code more quickly and easily, which should ease maintenance. It should also perform better, as it's doing less.

It may be that we decide that the pattern used by the cohort handler is worth the extra complexity, or we may decide on a hybrid (or even new way entirely), but I think it worthwhile to decide on a common pattern for event handlers.

I don't think we need to decide on a pattern for this PR, as we can change the code relatively painlessly later (as opposed to say schema changes!), but I wanted to start the debate.